Remove hybrid and legacy editor, and related code

maxme opened the pull request:
Main changes in this PR:
- added back the throw of `EditorFragmentNotAddedException` in Gutenberg ([here](https://github.com/wordpress-mobile/WordPress-Android/pull/10368/files#diff-02f567e707c1df878dfea548cfec3da7L610)) and Aztec fragment ([here](https://github.com/wordpress-mobile/WordPress-Android/pull/10368/files#diff-ffbdd6cd0e03b7a13fcdb13666c6d8c3L638)). I'm not sure why it wasn't done that way from the beginning, but returning the empty string was quite dangerous IMO since the editor activity could then locally auto-save and overwrite content and title in some cases.
- Removed `getSpannedContent` function and `saveMediaFile` callback from `EditorFragmentAbstract` since it was only used in the legacy editor.
- Removed `setLocalDraft` from `EditorFragmentAbstract` which wasn't used (but still set in the editor).
- Removed the `VisualEditorOptionsReceiver` that was only used to enable the Hybrid editor from
- Removed refs to the visual editor in `AppPrefs` in b554efa - also removed a test that was disabling featured images in post settings when hybrid or aztec was disabled (note that the hybrid editor was enabled years ago in wpandroid) - see [here](https://github.com/wordpress-mobile/WordPress-Android/pull/10368/files#diff-2a05761a1946e0ddb314645e44ab3eb6L271)
- Fixed example app and tests in c7baf08  - there is a remaining issue with Gutenberg in the example app, but this we'll probably won't use it, I didn't want to spend more time on it (I think it's not even worth filling a new issue).

Ideally we would reorganize this library (maybe re-integrate it into wpandroid?), but this is out of scope for this PR.

Ping @aforcier because most of the removed code is related to the Hybrid editor.
Ping @hypest because Gutenberg + dependencies setup is a bit complex.

hypest commented:
Added @daniloercoli and @marecar3 to assist on the review, not sure if I'll have the chance to do this alone :)

marecar3 commented:
Hey @maxme, nice work here, good to see that we are removing unused editors 👍 
Please, can we update this branch with the latest `develop` as latest `develop` has newer `version code` version? Thanks.

marecar3 commented on libs/editor/WordPressEditor/build.gradle:
```diff
@@ -31,14 +33,17 @@ apply from: 'https://gist.githubusercontent.com/hypest/f526fe0775dedce0ce0133f14
         versionCode 13
         versionName "1.3"
         minSdkVersion 21
         targetSdkVersion 26
```
I know it's not part of changes on this PR, but what could be a reason for having `targetSdkVersion` 26 and `compileSdkVersion ` 28? hmm.

maxme replied:
Updated in f840d763d2310d4be0b8b03cffdf9bc34502a1d9

marecar3 commented:
Hey @maxme!

When I want to build the example app from `../libs/editor` with the command `./gradlew assembleDebug` I am getting this kind of error

`> Task :WordPressEditor:compileDebugJavaWithJavac FAILED
/Users/markosavic/Documents/Development/Automattic/WordPress-Android-AndroidX/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java:368: error: cannot access Fragment
        if (PermissionUtils.checkAndRequestCameraAndStoragePermissions(this,
                           ^
  class file for android.support.v4.app.Fragment not found
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error
`

marecar3 commented on libs/editor/example/src/main/java/org/wordpress/android/editor/example/EditorExampleActivity.java:
```diff
@@ -57,27 +61,58 @@
 
     @Override
-    public void onAttachFragment(Fragment fragment) {
+    public void onAttachFragment(androidx.fragment.app.Fragment fragment) {
```
I think that there is no need for a full package path as we already have import on the top of the file?

maxme replied:
Fixed in 0a7e0b6b6d

marecar3 commented on libs/editor/example/src/main/java/org/wordpress/android/editor/example/EditorExampleActivity.java:
```diff
@@ -57,27 +61,58 @@
     @Override
     public void onBackPressed() {
-        Fragment fragment =  getFragmentManager()
+        androidx.fragment.app.Fragment fragment =  getSupportFragmentManager()
```
I think that there is no need for a full package path as we already have import on the top of the file?

maxme replied:
Yep, right, I'll update this and the previous one and do a quick check to see if there are other occurences.

maxme replied:
Fixed in 0a7e0b6b6d (I couldn't find any other occurences)

marecar3 commented:
Generally looks fine. It seems that the main app (WPAndroid) isn't broken, which is good.

Maybe it would be good to add some Test notes in the description of this task so that people who are not familiar with old editors, know what should be the focus while testing?

Also, it would be probably good to split this kind of PR into smaller parts as it's not easy to follow all those changes.

daniloercoli commented:
Tested the Android app (also the QuickPress Widget!) and worked as expected!

maxme commented:
> When I want to build the example app from `../libs/editor` with the command `./gradlew assembleDebug` I am getting this kind of error

Can you clean up the build and try again? It works fine here:

```
$ cd WordPress-Android/libs/editor/
$ ./gradlew assembleDebug build test

> Task :example:lint
Ran lint on variant release: 15 issues found
Ran lint on variant debug: 15 issues found
Wrote HTML report to file:///Users/max/work/automattic/WordPress-Android/libs/editor/example/build/reports/lint-results.html
Wrote XML report to file:///Users/max/work/automattic/WordPress-Android/libs/editor/example/build/reports/lint-results.xml

> Task :WordPressEditor:lint
Ran lint on variant debug: 45 issues found
Ran lint on variant release: 45 issues found
Wrote HTML report to file:///Users/max/work/automattic/WordPress-Android/libs/editor/WordPressEditor/build/reports/lint-results.html
Wrote XML report to file:///Users/max/work/automattic/WordPress-Android/libs/editor/WordPressEditor/build/reports/lint-results.xml

BUILD SUCCESSFUL in 3s
108 actionable tasks: 2 executed, 106 up-to-date
```

maxme commented:
> Maybe it would be good to add some Test notes in the description of this task so that people who are not familiar with old editors, know what should be the focus while testing?

Some editor history, there was 4 different WYSIWYG editors in WPAndroid:
- The Legacy editor (enhanced EditText).
- The Hybrid editor (WebView based, using a javascript dependency called ZSSEditor).
- Aztec (enhanced++ EditText).
- Gutenberg (RN based, using Aztec for rich text editing).

When Aztec was fully deployed and considered stable, the legacy and hybrid editors were made unavailable for our users (the switch was removed from the App settings screen), but the code was remaining.

This PR removes the code that was made unaccessible for our users. It doesn't change UX or app behavior (besides the potential fix from `EditorFragmentNotAddedException`). Because of this, there is no user scenario to test. I see only one with data receivers:

Test 1: check receivers are off:

- Open this https://bia.is/s/eTCy/editor-links.html on your test device.
- Click the 4 links, make sure nothing happens (check adb logs).

You can eventually make sure there is no references to legacy or hybrid in the App Settings or Site Settings (but since the code is now removed, we would notice it).

I also fixed part of the example app in the editor so you could check that, but honestly this is not used and will likely never be used again (this wasn't building for at least months): Open AS, tap "File" -> "Open", pick the `build.gradle` under `libs/editor/`, build "Example" and make sure Aztec loads (Gutenberg doesn't, it's a known issue and won't be covered in this PR).

maxme commented:
@marecar3 this PR is ready for a second round.

marecar3 commented:
Will take a look at it, thanks @maxme!

jkmassel commented:
Howdy! The `13.3` code freeze is today, so I'm bumping this to `13.4`. If you'd like it to be part of the `13.3` release, please target the `release/13.3` branch, and DM me once it's merged so that I can include it as a new beta release!
